internal/sdkv2provider/provider.go

```diff
@@ -21,7 +21,7 @@ import (
 const (
 	MAXIMUM_NUMBER_OF_ENTITIES_REACHED_SUMMARY = "You've attempted to add a new %[1]s to the `terraform-plugin-sdkv2` which is no longer considered suitable for use."
 	MAXIMUM_NUMBER_OF_ENTITIES_REACHED_DETAIL  = "Due the number of known internal issues with `terraform-plugin-sdkv2` (most notably handling of zero values), we are no longer recommending using it and instead, advise using `terraform-plugin-framework` exclusively. If you must use terraform-plugin-sdkv2 for this new %[1]s you should first discuss it with a maintainer to fully understand the impact and potential ramifications. Only then should you bump %[2]s to include your %[1]s."
-	MAXIMUM_ALLOWED_SDKV2_RESOURCES            = 108
```

Comment:
given that your new resource has two booleans that have functionality tied to `true` and `false`, i highly advise against doing this with SDKv2. you should do this with the framework instead else encounter zero value issues.